pp_netlib/network.py

```diff
@@ -235,10 +236,58 @@ def get_summary(self, print_to_std = True, summary_file_prefix = None):
                 summary.write(summary_contents)
             summary.close()
 
-    def visualize(self):
-        # code for calling viz functions
-        print("visualizing network")
-        return #files associated with viz
+    def visualize(self, viz_type, out_prefix, external_data = None):
+        from pp_netlib.functions import generate_mst_network, save_graph
+
+        self.mst_network = generate_mst_network(self.graph, self.backend)
```

Comment:
This should only be called if viz_type is mst, so put in conditional below

Reply:
in the original code, outputs_for_cytoscape produced an mst.graphml file if an mst network was passed to it. Should I therefore leave this as is or is it better to never save the mst when viz_type="cytoscape"?

Reply:
I think this is mixing up two things: visualisation and MST generation. The cytoscape part can also save the mst, but only if it's already in the object

Reply:
made it so that mst is only created when viz_type="mst"; cytoscape part now has an if statement that saves mst if it exists